src/hotspot/share/nmt/memMapPrinter.cpp

```diff
@@ -127,11 +127,21 @@ class CachedNMTInformation : public VirtualMemoryWalker {
   // Given a vma [from, to), find all regions that intersect with this vma and
   // return their collective flags.
   MemFlagBitmap lookup(const void* from, const void* to) const {
+    assert(from <= to, "Sanity");
+    // We optimize for sequential lookups. Since this class is used when a list
+    // of OS mappings is scanned (VirtualQuery, /proc/pid/maps), and these lists
+    // are usually sorted in order of addresses, ascending.
+    static uintx last = 0;
+    if (to <= _ranges[last].from) {
+      // not sequential? restart at 0
```

Comment:
I would prefer to say something like "the range is to the right of the given section, we need to re-start the search"

Reply:
okay